mindsdb/integrations/handlers/xgboost_handler/xgboost_handler.py

```diff
@@ -0,0 +1,471 @@
+
+
+class BoosterType(Enum):
+    gbtree = "gbtree"
+    gblinear = "gblinear"
+
+
+class ObjectiveType(Enum):
+    reg_linear = "reg:linear"
+    reg_squarederror = "reg:squarederror"
+    multi_softmax = "multi:softmax"
+    multi_softprob = "multi:softprob"
+
+
+class TaskType(Enum):
+    auto = "auto"
+    regressor = "regressor"
+    classifier = "classifier"
+    a = "a"
+    c = "c"
+    r = "r"
+    reg = "reg"
+    cla = "cla"
+
+
+def enum_to_str(type_class: Enum) -> str:
+    all = []
+    for element in type_class:
+        all.append(element.name)
+    return "|".join(all)
+
+
+def to_ts_dataframe(df: pd.DataFrame, time_col=None) -> str:
+    columns = list(df.columns.values)
+    if time_col is not None:
+        if time_col not in columns:
+            raise Exception("invalid column name: " + time_col)
+        if df[time_col].dtype != np.datetime64:
+            try:
+                df.index = pd.to_datetime(df[time_col])
+            except Exception as e:
+                raise Exception("can not convert column to datetime: " + time_col + " " + str(e))
+    else:
+        datetime_cols = list(df.select_dtypes(include=["datetime"]).columns.values)
+        if len(datetime_cols) > 0:
+            time_col = datetime_cols[0]
+            df.index = pd.to_datetime(df[time_col])
+        if time_col is None:
+            raise Exception("can not find datetime column for time series")
+    df.drop(columns=[time_col], inplace=True)
+    return time_col
+
+
+def check_target_column_type(df: pd.DataFrame, target: str, task_type: TaskType):
+    MAX_LABEL_NUMBER = 100
+    UNIQUE_CNT_TO_BE_REG = 5
+    target_type = df[target].dtype
+    label_cnt = len(df[target].unique())
+    if label_cnt == 1:
+        raise Exception("Label error: the count of unique value in target column can not be less than 2.")
+    if task_type == TaskType.regressor and (target_type == np.dtype("float") or target_type == np.dtype("int") or
+                                            target_type is np.dtype("float") or target_type is np.dtype("int")):
+        return TaskType.regressor, 0
+    if task_type == TaskType.classifier and (target_type == np.dtype("object") or target_type == np.dtype("bool") or
+                                             target_type is np.dtype("object") or target_type is np.dtype("bool")):
+        return TaskType.classifier, label_cnt
+
+    if task_type == TaskType.classifier and label_cnt <= MAX_LABEL_NUMBER:
+        return TaskType.classifier, label_cnt
+    if task_type == TaskType.auto:
+        if target_type == np.dtype("float") or target_type == np.dtype("int") \
+                or target_type is np.dtype("float") or target_type is np.dtype("int"):
+            if label_cnt <= UNIQUE_CNT_TO_BE_REG:
+                return TaskType.classifier, label_cnt
+            else:
+                return TaskType.regressor, 0
+        else:
+            return TaskType.classifier, label_cnt
+
+
+class FeatureProcessor:
```

Comment:
Just FYI, we are working on a package (`dataprep_ml`) that will be handler-agnostic and may help here once it's fully released.

Reply:
We could add support for `category_encoders` there, but this is fine for now.